Sources/Security.php

```diff
@@ -925,6 +926,9 @@ function allowedTo($permission, $boards = null, $any = false)
 	elseif (!is_array($boards))
 		$boards = array($boards);
 
+	if (isset($prem_cache[hash('md5', $user_info['id'].implode($permission).implode($boards).$any)]))
```

Comment:
Here and throughout, please use delimiters between each component of the hashed strings, including between the imploded array elements, like this:
```
return $prem_cache[hash('md5', $user_info['id'] . '-' . implode(',', $permission) . '-' . implode(',', $boards) . '-' . $any)];
```
This will prevent possible collisions if, say, $boards contains `12, 32` in one case and `1, 2, 32` in another case.